app/assets/javascripts/application.js

```diff
@@ -15,8 +15,33 @@
 //= require foundation
 //= require_tree .
 
-$(document).foundation();
+$(function(){ $(document).foundation(); });
 
 $(function() {
```

Comment:
I'd suggest this line, and all the new JS below it you probably want to move out into another .js file that the application.js file requires. Try to name that file to describe what pages or area of the app is being affected by the JS it contains so it'll be easier to find/understand the next time you look at it.